### ISSUE TITLE
"Create a question" missing headers

### ISSUE THREAD

**Issue by glorialiou** [civiform/civiform#1914]
Certain sections of the "Create a question" page are missing a header (e.g., the export options). The header is needed to visually chunk the various sections and to make this page more understandable/easy to use. It could also be helpful to revise UI copy to add clarity about what the different export options mean.

**Comment by cassandralfernandes** [comment]
Mocks here: https://docs.google.com/presentation/d/15fcq3dYFgy1Dfy2UjZybmHD9KIW5cz8Q1BjwgOpLJ7g/edit#slide=id.g13fc01115e1_0_0

**Comment by clouser2** [comment]
Just to confirm, this issue covers adding headers for export options and the other question subsections (similar to program). The mocks referenced are more around changing the text for the export options, correct?

Just want to make sure that this is the intention of linking the mocks and not that we're switching to the UI within the mocks as part of this work.

**Comment by cassandralfernandes** [comment]
Correct, we are not switching to the UI within the mocks as part of this work. The header on the mock "data privacy setting" is where we landed. 

We also do want to add clarity around what each setting means, although I'm not sure I could articulate succinctly what each setting refers to. WDYT, @clouser2?  @rahaghas @jessborland any proposed help text to describe the export options?

**Comment by clouser2** [comment]
A proposed solution. For reference, here's the current question UI:

![Image](https://user-images.githubusercontent.com/1870301/184927951-4bd0d00b-f4c3-44de-9a8c-74bca26ebaf7.png)

Note that depending on the question type, additional options emerge. (minimum / maximum for numbers, option choices for dropdowns, etc). It likely makes sense to keep those in the existing "Question settings" section visually.

Current field order:
  * Question enumerator (when selected, shows help text explaining that $this needs to be used within the question's text and help text)
  * Name - Internal name (not currently modifiable)
  * Description - Internal description
  * Question text - External-facing question name
  * Question help text - External-facing question help text
  * Optional question settings. Within a "Question settings" field.
  * Export options

A proposed reorganization of the fields and grouping follows:
  * "Internal question information" section. Subtext "This will only be visible to administrators" (similar to the program page)
    - Administrative name
    - Description
  * "Public question information" section. Subtext "This will be visible to the public" (similar to program page)
    - Question text
    - Question help text
  * "Question settings" section (same as current)
    - Existing per-question-type settings
  * "Data privacy setting" section. Has a radio with the following options:
    - Don't allow answers to be exported
    - Export exact answers
    - Export obfuscated answers

@cassandralfernandes @rahaghas , can you provide feedback on 1) the proposed field ordering / sectioning / naming and 2) Whether the proposed export options make sense? Note that the mocks didn't seem to have an option for exporting obfuscated answers.

If it's easiest, I'm fine with you just editing my comment with any proposed changes y'all would like to make.

I'll wait to start on this until we've got it ironed out. We'll likely want to wait on #3023 to be completed before starting this anyway, since it's touching code in the same area.

**Comment by rahaghas** [comment]
Is it out of scope of this buf to combine Administrative name and Description into one field called Admin note?

Change "Internal question information" and Subtext "This will only be visible to administrators" to simply "Visible to administrators only"
Change "Public question information" to "Visible to applicants"

Change "Data privacy setting" to "Data export settings"
"Don't allow answers to be exported" sounds the same as "export obfuscated answers", I'm not sure what the difference is.

**Comment by clouser2** [comment]
> Is it out of scope of this buf to combine Administrative name and Description into one field called Admin note?

Yes, out of scope. Administrative name is the internal identifier. We have a separate issue (#3075) for trying to "remove" the need for internal names or make them a bit less prevalent in the UI. I'd propose we start by organizing the information here and we can yank it out once that work is addressed.

> Change "Internal question information" and Subtext "This will only be visible to administrators" to simply "Visible to administrators only" Change "Public question information" to "Visible to applicants"

Sounds good. Do you want me to do the same for the corresponding program section and subtext?

> Change "Data privacy setting" to "Data export settings" "Don't allow answers to be exported" sounds the same as "export obfuscated answers", I'm not sure what the difference is.

Works for me. So this toggle controls whether data will be exported in the demography CSV. Using the current option names, I believe they work as follows:
  * No export - the answer won't appear in any way in the exported data
  * Export Value - the raw answer as provided by the applicant will appear in the exported data
  * Export Obfuscated - an "obfuscated" answer will appear in the exported data. Obfuscated essentially means that the "raw" answer text won't appear in the data, but a muddled version of it. @bion should be able to provide a bit more info here.

**Comment by rahaghas** [comment]
Yes let's do the same for the corresponding program section and subtext

Ok yeah I think we need a better explanation for Export obfuscated, and I think I need to better understand exactly how it exports

**Comment by bion** [comment]
In the demography export only (the CiviForm admin's export feature, not the Program admin's), answers to "export obfuscated" questions are converted to string and [SHA256 hashed](https://github.com/civiform/civiform/blob/main/server/app/services/export/CsvExporter.java#L139-L147) using the the application secret (the value for `play.http.secret.key` config key) as a salt.

UI text describing this could maybe something along the lines of:

> Export obfuscated will cause the applicant's answer to this question to be cryptographically obfuscated, exporting text that is unique to the applicant's answer but does not reveal what the original text was. It is not possible to derive the applicant's original input from the resulting exported value. Only other questions with the exact same answer will have the same exported value. For example, a social security number question:
> Input: 123-45-6789 Example output: d158596dd5a6cae6fcb282832885631553ecc8c8b0bf07f84c4aa691953cd0da
> Input: 123-45-6780 Example output: 01ba4719c80b6fe911b091a7c05124b64eeece964e09c058ef8f9805daca546b

**Comment by clouser2** [comment]
@rahaghas For privacy settings, how about we continue to use the "obfuscated" text and have the options be:
Don't allow answers to be exported
Export exact answers
Export obfuscated answers

Then, we can put a "Help text" icon next to the "Data privacy settings" label and pop a modal explaining in more detail what obfuscated means (even potentially lifting the entire blurb that @bion metnioned above)?

For now, I'm going to perform all of the other changes described here and come back to data privacy settings once we've resolved this issue.

**Comment by clouser2** [comment]
Actually, for the first pass, I'll leave the demographic options exactly as currently configured. We can address changing the text once we've figured out the final text.

**Comment by cassandralfernandes** [comment]
The proposed data export options (Don't allow answers to be exported; Export exact answers; Export obfuscated answers) sound fine, especially with the pop up modal with more info. Do we use those anywhere else in product? 

If we can't add the modal right away, we can also use the subtext under the export settings heading to link out to a civiform docs article with info about each option (e.g. Read more about each of the data export options here). I'm adding info on export options to the [manage questions article](https://docs.civiform.us/user-manual/civiform-admin-guide/manage-questions) today anyway

**Comment by clouser2** [comment]
Nice! Linking to the external docs seems better to me, since we'll have a lot more room to explain things there. And it'll be less chance for our in-app docs to drift out of date with what you're writing.

**Comment by cassandralfernandes** [comment]
Cool, let's do that then. Article is live here: https://docs.civiform.us/user-manual/civiform-admin-guide/manage-questions#question-export-settings 

If anyone has thoughts on the text, free to edit it directly or ping me and I can make changes
